cmake/projects/libsodium/schemes/url_sha1_libsodium_autogen_autotools.cmake.in

```diff
@@ -0,0 +1,111 @@
+# Copyright (c) 2013, 2015 Ruslan Baratov, Alexandre Pretyman
+# All rights reserved.
+
+cmake_minimum_required(VERSION 3.0)
+project(Hunter)
+
+list(APPEND CMAKE_MODULE_PATH "@HUNTER_SELF@/cmake/modules")
+
+include(hunter_libsodium_install_tools)
+include(hunter_user_error)
+include(hunter_status_debug)
+include(hunter_test_string_not_empty)
+include(hunter_autotools_project)
+
+hunter_status_debug("Scheme: url_sha1_libsodium_autogen_autotools")
+
+if("@MSVC@")
+    hunter_user_error("Autotools scheme not supported with Visual Studio")
+endif()
+
+# Check preconditions
+hunter_test_string_not_empty("@HUNTER_SELF@")
+hunter_test_string_not_empty("@HUNTER_EP_NAME@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_URL@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_SHA1@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_DOWNLOAD_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_SOURCE_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_BUILD_DIR@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_CONFIGURATION_TYPES@")
+hunter_test_string_not_empty("@HUNTER_PACKAGE_INSTALL_PREFIX@")
+hunter_test_string_not_empty("@HUNTER_INSTALL_PREFIX@")
+
+# Autotools Project
+foreach(build_config_type @HUNTER_PACKAGE_CONFIGURATION_TYPES@)
+
+    # Check for valid configuration
+    string(COMPARE EQUAL "${build_config_type}" "Release" is_release)
+    string(COMPARE EQUAL "${build_config_type}" "Debug"   is_debug)
+    if(NOT (is_release OR is_debug))
+        hunter_user_error("libsodium: Invalid build configuration type provided. Valid values are: \"Release\", \"Debug\", \"Release\;Debug\" (HUNTER_PACKAGE_CONFIGURATION_TYPES)")
+    endif()
+
+    hunter_autotools_project(@HUNTER_EP_NAME@${build_config_type}
```

Comment:
probably this is a source of a race condition, all projects should depends on each other since they all do download step (download will be done once by first project, others will do check sha1 only). see for example: https://github.com/ruslo/hunter/blob/83b35ac72b59cce39bfad23ac81e04b4d365868f/cmake/schemes/url_sha1_cmake.cmake.in#L140